File: src/Lesson6/Java/Ex5.java
package Lesson6.Java;

public class Ex5 {
    public static void main(String[] args) {

        int[] arr = {4, 3, 2, 1};

        int swaps = countMinSwaps(arr);

        System.out.println(swaps);
    }

    public static int countMinSwaps(int[] arr) {

        int swaps = 0;

        for (int i = 0; i < arr.length; i++) {

            int minIndex = i;

            for (int j = i + 1; j < arr.length; j++) {

                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }

            if (minIndex != i) {

                int temp = arr[i];
                arr[i] = arr[minIndex];
                arr[minIndex] = temp;

                swaps++;
            }
        }

        return swaps;
    }
}

File: src/Lesson6/forkotlin/Ex5.kt
package Lesson6.forkotlin

fun main() {
    val arr = intArrayOf(4, 3, 2, 1)

    val swaps = countMinSwaps(arr)

    println(swaps)
}

fun countMinSwaps(arr: IntArray): Int {
    var swaps = 0

    for (i in arr.indices) {
        var minIndex = i

        for (j in i + 1 until arr.size) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j
            }
        }

        if (minIndex != i) {
            val temp = arr[i]
            arr[i] = arr[minIndex]
            arr[minIndex] = temp

            swaps++
        }
    }

    return swaps
}